### PR TITLE
Pro-artistry: Add a callback for the Rotated Shadow Map technique.

### PR DESCRIPTION
**Pull request by YohannVaastUnity**
/!\ This PR is a draft, just a proposition for the HDRP team. /!\ 

Reminder of the `Rotated Shadow Map` technique:
https://drive.google.com/file/d/18_Px85w9-_VxRXKBoo1UnVy1OUpN1Ri_/view?usp=sharing

Here's a link to the prototype we developed. Also, this technique has been used in several Disney projects (Big Hero / Baymax) and Sherman (1 & 2), proving it's efficiency. 
https://github.com/YohannVaastUnity/HDRP_RotatedShadowMap/pull/1

Here's a quick gif of the accumulation process:
RSM

**Regarding the PR**

```Implementation
- The callback is used to change the View matrix (of the Spotlight shadow request) outside of the HDRP package. In the future, it will be integrated in the Recorder package (so the Recorder package will use this callback).
- We are rotating this matrix (based on the `visibleLight.localToWorldMatrix`), applying our RSM technique.
- It's as simple as that.
- I don't know if HDRP has a norm regarding callback or anything else, explaining why this PR is a draft.
```
 Thanks!

### CODE REVIEW

**Review by clusty — APPROVED** [review]
Looks simple enough for me

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs [hunk 1]
```diff
@@ -1979,6 +1979,9 @@ public int areaLightEmissiveMeshLayer
             }
         }
 
+        public delegate Matrix4x4 CustomViewCallback(Matrix4x4 lightLocalToWorldMatrix);
+        public CustomViewCallback CustomViewCallbackEvent;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs:1983]: please add documentation for public API :)

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs:1983]: + add a changelog

Comment (YohannVaastUnity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs:1986]: Added :)

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
triggered formatting job: https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/prototype%252Frotated-shadow-map-callback/.yamato%252F_formatting.yml%2523formatting_apply/10517599/job/pipeline

**Pull request merged**